packages/qdrant-loader-core/src/qdrant_loader_core/llm/providers/gemini.py

```
@@ -0,0 +1,350 @@
+                pass
+
+            embeddings = getattr(response, "embeddings", None) or []
+            vectors = [list(getattr(item, "values", []) or []) for item in embeddings]
+            if len(vectors) != len(inputs):
+                raise ServerError(
+                    f"Gemini embed_content returned {len(vectors)} embeddings "
+                    f"for {len(inputs)} inputs (expected 1:1). "
+                    f"Verify the model id {self._model!r} supports per-input embeddings."
+                )
+            return vectors
+        except Exception as exc:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)